doc/man3/EVP_MAC.pod

```diff
@@ -0,0 +1,235 @@
+
+B<EVP_MAC_CTX> is a context type that holds internal information used
+to compute MACs from the input.
+
+=head2 Context manipulation functions
+
+EVP_MAC_CTX_new() creates a new context for the MAC type C<mac>.
+EVP_MAC_CTX_new_id() creates a new context for the numerical MAC
+identity <nid>.  The created context can then be used with most other
+functions described here.
+
+EVP_MAC_CTX_free() frees the contents of the context as well as the
+context itself.  B<NULL> is a valid parameter, for which this function
+is a no-op.
+
+EVP_MAC_CTX_copy() makes a deep copy of the C<src> context to the
+C<dest> context.  The C<dest> context I<must> have been created before
+calling this function.
+
+EVP_MAC_CTX_mac() returns the B<EVP_MAC> associated with the context
+C<ctx>.
+
+=head2 Computing functions
+
+EVP_MAC_init() sets up the context with information given through
+diverse controls.  Implementations may vary on this, some of them may
+do their initialization in the controls themselves.  However, this
+should be called at the very least before calling EVP_MAC_update() and
+EVP_MAC_final().
+
+EVP_MAC_reset() resets the computation for the given context.  This
+may not be supported by the underlying implementation.
+
+EVP_MAC_update() adds C<datalen> bytes from C<data> to the MAC input.
+
+EVP_MAC_final() does the final computation and stores the result in
+the memory pointed at by C<out>, and sets its size in the B<size_t>
+the C<poutlen> points at.  If C<out> is B<NULL>, then no computation
+is made.  To figure out what the output length will be and allocate
+space for it dynamically, simply call with C<out> being B<NULL> and
+C<poutlen> pointing at a valid location, then allocate space and make
+a second call with C<out> pointing at the allocated space.
+
+EVP_MAC_ctrl() is used to give the MAC associated with the given
+context extra data that can't be easily given any other way.  This
+includes the MAC key, and for MACs that use other algorithms to do
+their computation, this is also the way to tell it which one to use.
+This functions takes variable arguments, the exact expected arguments
+depend on C<cmd>.  See </CONTROLS> below for a description of standard
+controls.
+
+EVP_MAC_ctrl_str() is an alternative to EVP_MAC_ctrl() to control the
+MAC implementation as E<lt> C<type>, C<value> E<gt> pairs.  The MAC
+implementation documentation should specify what control type strings
+are accepted.
+
+EVP_MAC_str2ctrl() and EVP_MAC_hex2ctrl() are helper functions to
+control the MAC implementation with raw strings or with strings
+containing hexadecimal numbers.  The latter are decoded into
+bitstrings that are sent on to EVP_MAC_ctrl().
+
+=head2 Information functions
+
+EVP_MAC_size() returns the MAC output size for the given context.
+
+EVP_MAC_nid() return the numeric identity of the given MAC implementation.
+
+EVP_MAC_name() return the numeric identity of the given MAC implementation.
```

Comment:
I'm going to guess it returns a name, not a number.

Reply:
Yes.  Will fix.

Reply:
Fixed